commands/provider/provider.go

```diff
@@ -10,13 +10,17 @@ type Provider struct {
 	*CoreProvider
 	*ServiceProvider
 	*WalletProvider
+	*MarketplaceProvider
 }
 
 // New creates Provider based on given CoreClient.
 func New(c coreapi.CoreClient, d daemon.Daemon) *Provider {
+	wp := NewWalletProvider(c)
+	mp := NewMarketplaceProvider(c)
```

Comment:
`NewMarketplaceProvider` could have the `NewWalletProvider` as dependency to have the list of addresses in select choice

Reply:
What do you mean by `in select choice`? You mean in the commands when the flag `--account` is not provided?

Reply:
yes we could have the list of addresses from the provider and just ask the user to select the address instead of fill it

Reply:
Yes, i wanted to do it but I didn't yet..
But anyway, putting `WalletProvider` in `MarketplaceProvider` will not help for this.
The command should ask for it in the `preRun` function, so before calling the provider.

Reply:
I would do in another PR